### PR TITLE
feat: fill in now() for completed date for audit: ENT-4918

### PR DESCRIPTION
**Pull request by binodpant**
if audit course completed, and no date from api, then fill in with now() for completed audit courses
this is needed since we get None completed_date for audit courses in some cases when we decide to call it completed based on incomplete_count == 0

JIRA: https://openedx.atlassian.net/browse/ENT-4918

if for some rason a date does come back from api it still honors it
if audit complete is not completed (based on incomplete_count), it still avoid filling in
if track isn't even audit, it avoid filling date in

so only sets date to now() when audit + completed + date_not_coming_from_api

ENT-4918

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/integrated_channel/exporters/learner_data.py [hunk 1]
```diff
@@ -312,6 +327,10 @@ def get_incomplete_content_count(self, enterprise_enrollment, channel_name):
         '''
         incomplete_count = None
         is_audit_enrollment = enterprise_enrollment.is_audit_enrollment
+
+        # The decision to not get incomplete count for non audit enrollments can be questioned
+        # Right now we don't use this number for non audit. But this conditoin can be just removed
```
Comment (alex-sheehan-edx) [inline integrated_channels/integrated_channel/exporters/learner_data.py:332]: `condition`?

File: integrated_channels/integrated_channel/exporters/learner_data.py [hunk 2]
```diff
@@ -269,13 +269,17 @@ def get_grades_summary(
         course_details,
         enterprise_enrollment,
         channel_name,
+        incomplete_count=None,
```
Comment (alex-sheehan-edx) [inline integrated_channels/integrated_channel/exporters/learner_data.py:272]: Remind me who the consumer of this is and why it's getting added?

**Review by alex-sheehan-edx — APPROVED** [review]
LGTM other than one minor comment nit and a context question!

File: integrated_channels/integrated_channel/exporters/learner_data.py (hunk 2 above)
Comment (binodpant) [reply integrated_channels/integrated_channel/exporters/learner_data.py:272]: it's used to decide when to fill in the date automatically, among other things, later in the method. for audit we only want to fill if course is complete based on `incomplete_count==0`

Comment (binodpant) [reply integrated_channels/integrated_channel/exporters/learner_data.py:272]: we don't use it for non audit right now, I think we can further refactor so audit based code is compeletely separate from non audit...

**Pull request merged**